docs/references/javascript/clerk/clerk.mdx

```diff
@@ -16,28 +16,79 @@ This is the main entrypoint class for the `@clerk/clerk-js` package. It contains
+| `user` | [`User`][user-ref] \| `null` \| `undefined` | A shortcut to `Session.user` which holds the currently active `User` object. If the session is `null` or `undefined`, the user field will match. |
+| `organization` | [`Organization`][organization-ref] \| `null` \| `undefined` | A shortcut to the last active `Session.user.organizationMemberships` which holds an instance of a `Organization` object. If the session is `null` or `undefined`, the user field will match. |
 | `publishableKey` | `string \| undefined` | The publishable key from your Clerk Dashboard, used to connect to Clerk |
 
-## Methods
+## `Clerk` methods
+
+### `addListener()`
+
+Registers a listener that triggers a callback whenever a change in the [`Client`][client-ref], [`Session`][session-ref], or [`User`][user-ref] object occurs. This method is primarily used to build frontend SDKs like [`@clerk/clerk-react`](https://www.npmjs.com/package/@clerk/clerk-react).
+
+```typescript
+function addListener(listener: ((emission: Resources) => void)): UnsubscribeCallback;
+```
+
+#### `Resources`
+
+| Name | Type |
+| --- | --- |
+| `client` | [`Client`][client-ref] |
+| `session` | [`Session`][session-ref] \| `null` \| `undefined` |
+| `user` | [`User`][user-ref] \| `null` \| `undefined` |
+| `organization` | [`Organization`][organization-ref] \| `null` \| `undefined` |
+| `lastOrganizationInvitation` | [`OrganizationInvitation`](/docs/references/javascript/organization-invitation) \| `null` \| `undefined` |
+| `lastOrganizationMember` | [`OrganizationMembership`](/docs/references/javascript/organization-membership) \| `null` \| `undefined` |
+
+<Callout>
+  Please note that the `Session` and `User` object have a special relationship that the type definition alone does not capture:
+    - When there is an active session, `user === session.user`. When there is no active session, `User` and `Session` will both be `null`.
+    - When a session is loading, `User` and `Session` will be `undefined`.
+</Callout>
+
+#### `addListener()` returns
+
+The `addListener` method returns an `UnsubscribeCallback` function that can be used to remove the listener from the `Clerk` object.
+
+```typescript
+type UnsubscribeCallback = () => void;
+```
+
+### `authenticateWithMetamask()`
+
+```typescript
+function authenticateWithMetamask({
+  redirectUrl,
+  signUpContinueUrl,
+  customNavigate,
+}?: AuthenticateWithMetamaskParams): Promise<void>;
+```
+
+#### `AuthenticateWithMetamaskParams`
+
+| Name | Type | Description |
+| --- | --- | --- |
+| `redirectUrl?` | `string \| undefined` | Full URL or path to navigate after successful sign in or sign up. |
+| `signUpContinueUrl?` | `string \| undefined` | The location to continue navigation to in the sign up process if data is missing. |
+| `customNavigate?` | `((to: string) => Promise<unknown> \| unknown) \| undefined` | Allows you to define a custom navigation function |

```

Comment:
Probably out of scope of this PR but would love a little indication as to why you'd want to define a custom navigation function, similar to how we explain `signUpContinueUrl` is for if data is missing

Reply:
Ooo can you add a Linear ticket for that!